### PR TITLE
correctly store origin dst addr in map_of_dst_info at ipv4-mapped ipv…

### LINKED ISSUE #756: waypoint failed to process ipv4-mapped ipv6 connections
**What happened**:

In default k8s environment (non pure ipv6 env), java applications will use ipv4-mapped ipv6 to create connections by default. 

In the absence of waypoint, Kmesh has alreay support this scenario, ref: https://github.com/kmesh-net/kmesh/issues/291

But it still fails when there is a waypoint.

**What you expected to happen**:

Kmesh could handle both with and without waypoint.

**How to reproduce it (as minimally and precisely as possible)**:

Deploy bookinfo where `reviews` service is a java application. When it accesses the `ratings` service, it would create ipv4-mapped ipv6 connection. Then deploy waypoint for svc `ratings`, ref: https://kmesh.net/en/docs/userguide/install_waypoint/

Access bookinfo and Kmesh daemon will output the following error log:

```bash
time="2024-08-20T08:33:55Z" level=info msg="can't find service correspond workload: ratings-svc-waypoint-5d86d48c55-flxrd" subsys=pkg/telemetry
time="2024-08-20T08:33:55Z" level=info msg="[KMESH] DEBUG: origin addr=[10:0a60:4cfe:0000:0000:0000:0000:0000:0000:9080]\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[KMESH] DEBUG: bpf find frontend addr=[10:0a60:4cfe:0000:0000:0000:0000:0000:0000:9080]\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[SERVICE] DEBUG: find waypoint addr=[0a60:c461:0000:0000:0000:0000:0000:0000:15008]\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[KMESH] DEBUG: enter cgroup/connect6\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[KMESH] DEBUG: origin addr=[10:0a60:c461:0000:0000:0000:0000:0000:0000:15008]\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[KMESH] DEBUG: bpf find frontend addr=[10:0a60:c461:0000:0000:0000:0000:0000:0000:15008]\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[BACKEND] DEBUG: access the backend by service:2317460390\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[BACKEND] DEBUG: get the backend addr=[0af4:01db:0000:0000:0000:0000:0000:0000:15019]\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[KMESH] DEBUG: enter cgroup/connect6\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[KMESH] DEBUG: origin addr=[10:0af4:01db:0000:0000:0000:0000:0000:0000:15019]\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[KMESH] DEBUG: bpf find frontend addr=[10:0af4:01db:0000:0000:0000:0000:0000:0000:15019]\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[SOCKOPS] ERR: enable encoding metadata failed!, err is -16" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="[SENDMSG] DEBUG: get valid dst, do encoding...\n" subsys=ebpf
time="2024-08-20T08:33:55Z" level=info msg="get destination service host failed" subsys=pkg/telemetry
time="2024-08-20T08:33:55Z" level=info msg="can't find service correspond workload: ratings-svc-waypoint-5d86d48c55-flxrd" subsys=pkg/telemetry
time="2024-08-20T08:33:55Z" level=info msg="get destination service host failed" subsys=pkg/telemetry
time="2024-08-20T08:33:55Z" level=info msg="can't find service correspond workload: reviews-svc-waypoint-6884756fc5-wvdvh" subsys=pkg/telemetry
time="2024-08-20T08:33:55Z" level=info msg="get destination service host failed" subsys=pkg/telemetry
time="2024-08-20T08:33:55Z" level=info msg="can't find service correspond workload: reviews-svc-waypoint-6884756fc5-wvdvh" subsys=pkg/telemetry
```

**Anything else we need to know?**:

**Environment**:
- Kmesh version:
- Others:

### PR DESCRIPTION
**Pull request by kwb0523**
…6 scene

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
In waypoint_manager func, the ipv6 addr in kmesh_ctx->orig_dst_addr.ip6 is reversed to ipv4 format, and this format of ipv6 address inapplicable for later waypoint processing, it need store the origin dst addr directly.

**Which issue(s) this PR fixes**:
Fixes #756

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by YaoZengzeng** [comment]
Kmesh daemon will fail to restart:

```bash
root@kurator-linux-0002:~/kmesh# kklp kmesh-8ktvh 
none on /sys/fs/bpf type bpf (rw,relatime)
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --bpf-fs-path=\"/sys/fs/bpf\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --cgroup2-path=\"/mnt/kmesh_cgroup2\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --cni-etc-path=\"/etc/cni/net.d\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --conflist-name=\"\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --enable-bpf-log=\"true\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --enable-bypass=\"false\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --enable-mda=\"false\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --enable-secret-manager=\"false\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --help=\"false\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --mode=\"workload\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="FLAG: --plugin-cni-chained=\"true\"" subsys=manager
time="2024-08-23T07:22:17Z" level=info msg="oldGitVersion: 2800172400 newGitVersion: 2800172400" subsys=pkg/bpf
time="2024-08-23T07:22:17Z" level=info msg="kmesh start with Restart, load bpf maps and prog from last" subsys=pkg/bpf
time="2024-08-23T07:22:17Z" level=info msg="kmesh restart, not clean bpf map and prog" subsys=pkg/bpf
Error: bpf Load failed, field CgroupConnect4Prog: program cgroup_connect4_prog: load program: permission denied: 404: (7b) *(u64 *)(r10 -56) = r1: 405: (79) r6 = * (truncated, 278 line(s) omitted)
time="2024-08-23T07:22:17Z" level=error msg="bpf Load failed, field CgroupConnect4Prog: program cgroup_connect4_prog: load program: permission denied: 404: (7b) *(u64 *)(r10 -56) = r1: 405: (79) r6 = * (truncated, 278 line(s) omitted)" subsys=main
kmesh exit
```

File: bpf/kmesh/workload/include/backend.h [hunk 1]
```diff
@@ -28,6 +28,8 @@ static inline int waypoint_manager(struct kmesh_context *kmesh_ctx, struct ip_ad
         value_tuple.ipv4.daddr = kmesh_ctx->orig_dst_addr.ip4;
         value_tuple.ipv4.dport = ctx->user_port;
     } else if (ctx->family == AF_INET6) {
+        if (!is_ipv4_mapped_addr(kmesh_ctx->orig_dst_addr.ip6))
```
Comment (YaoZengzeng) [inline bpf/kmesh/workload/include/backend.h:31]: The return type is of type bool, so remove "!"?

**Review by hzxuzhonghu — COMMENTED** [review]
Should add an e2e test with bookinfo example and add namespace waypoint for each service

File: bpf/kmesh/workload/include/backend.h (hunk 1 above)
Comment (hzxuzhonghu) [reply bpf/kmesh/workload/include/backend.h:31]: how can waypoint process ipv6(v4 mapped format)?

Comment (YaoZengzeng) [reply bpf/kmesh/workload/include/backend.h:31]: waypoint could correctly handle pure ipv4, pure ipv6 and ipv4-mapped ipv6 address, ref: https://github.com/envoyproxy/envoy/blob/main/source/common/network/address_impl.cc#L87 

As long as the address encoded in tlv is valid.

**Comment by YaoZengzeng** [comment]
> Should add an e2e test with bookinfo example and add namespace waypoint for each service

ref: #766

**Pull request merged**